### ISSUE TITLE
Rule Change: max-nested-callbacks should count functions passed to a constructor call

### ISSUE THREAD

**Issue by DMartens** [eslint/eslint#20998]
### What rule do you want to change?

max-nested-callbacks

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
new Promise(() => {});
new MutationObserver(() => {});
```

### What does the rule currently do for this code?

Does not count the function passed to the constructor call as a callback.
I would count any function passed to a function (constructors are "special" functions) to be invoked later a callback.
For the added option I think an appropriate name would be `checkConstructorCallCallbacks` (default `false`).

### What will the rule do after it's changed?

Count functions passed directly to a function as a callback if the option is set.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

I originally thought of this change as a bugfix (see [PR](#20979)).

**Comment by mdjermanovic** [comment]
Makes sense to me. @eslint/eslint-team thoughts?

**Comment by derekblackwood61-lang** [comment]
My reasoning is that constructor calls like new Promise() and new
MutationObserver() receive functions that are effectively callbacks because
they're invoked later by the constructed object rather than immediately.
Since changing the default behavior could be breaking, I think introducing
an opt-in option such as checkConstructorCallCallbacks (defaulting to
false) would preserve backward compatibility while allowing stricter
enforcement for teams that want it.
I'm happy to put together a PR with the implementation, tests, and
documentation if the team agrees with this approach.

**Comment by Pixel998** [comment]
Sounds good to me as well. Marking this as accepted.
